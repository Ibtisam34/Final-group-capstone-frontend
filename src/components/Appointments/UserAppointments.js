import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Card from 'react-bootstrap/Card';
import { fetchAppointments } from '../../redux/appointments/appointment';
import { fetchDoctors } from '../../redux/doctors/doctorsReducer';
import SingleAppointment from './SingleAppointments';
import { deleteAppointment } from '../../redux/appointments/deleteAppointments';

const UserAppointments = () => {
  const dispatch = useDispatch();
  const doctors = useSelector((state) => state.doctorsReducer);
  const { appointments } = useSelector((state) => state.appointmentsReducer);
  const deletedRes = useSelector((state) => state.deleteAppointmentReducer);
  const userId = parseInt(localStorage.getItem('userId'), 10);

  useEffect(() => {
    dispatch(fetchDoctors());
    dispatch(fetchAppointments());
  }, [deletedRes, dispatch]);

  const handleDelete = (e, id) => {
    e.preventDefault();
    dispatch(deleteAppointment(id)).then(() => {
      dispatch(fetchAppointments());
    });
  };

  return (
    <>
      <h1 className="text-center m-4">appointments</h1>
      <div className="appointments-container">
        {
          appointments[0] && appointments[0].map((appointment) => (
            appointment.user_id === parseInt(userId, 10)
            && (
              <div key={appointment.id} className="appointment">
                <Card>
                  <Card.Header as="h5">
                    appointment #
                    {appointment.id}
                  </Card.Header>
                  <Card.Body>
                    <SingleAppointment
                      doctors={doctors}
                      appointment={appointment}
                      key={appointment.id}
                    />
                    <div className="d-flex justify-content-end">

                      <button type="submit" onClick={(e) => handleDelete(e, appointment.id)} className="btn btn-danger">Cancel the appointment</button>

                    </div>
                  </Card.Body>
                </Card>
              </div>
            )
          ))
        }
      </div>
    </>
  );
};

export default UserAppointments;
