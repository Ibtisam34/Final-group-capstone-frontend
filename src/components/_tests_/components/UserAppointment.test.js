import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../../../redux/configureStore';
import UserAppointments from '../../Appointments/Appointments';

describe('UserAppointment', () => {
  it('renders user appointment page correctly', () => {
    const tree = render(
      <Provider store={store}>
        <Router>
          <UserAppointments />
        </Router>
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });

  
});
