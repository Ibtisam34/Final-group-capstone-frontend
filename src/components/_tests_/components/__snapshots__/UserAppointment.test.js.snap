// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserAppointment renders user appointment page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="reservations-container"
      >
        <div
          class="reservation-header-links"
        >
          <a
            href="/undefined"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              style="color: rgb(13, 110, 253); font-size: 1.2rem; margin: 5px;"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
              />
            </svg>
          </a>
        </div>
        <div
          class="reservations-description"
        >
          <div
            class="reservations-header"
          >
            <h1>
              Book an Appointment
            </h1>
            <p>
              Select a doctor to make an appointment with.
            </p>
          </div>
          <select
            id="dropdown-menu-align-end"
            placeholder="Choose a Doctor you want to see."
            style="outline: none; width: 95%; border-radius: 2px;"
          >
            <option
              disabled=""
            >
              Choose a Doctor you want to see
            </option>
          </select>
          <div
            class="reservations-buttons"
          >
            <div
              class="react-datepicker-wrapper"
            >
              <div
                class="react-datepicker__input-container "
              >
                <span
                  aria-live="polite"
                  class="react-datepicker__aria-live"
                  role="alert"
                />
                <input
                  class=""
                  type="text"
                  value="04/26/2023"
                />
              </div>
            </div>
            <button
              class="book-btn"
              type="submit"
            >
              Book now
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="reservations-container"
    >
      <div
        class="reservation-header-links"
      >
        <a
          href="/undefined"
        >
          <svg
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            style="color: rgb(13, 110, 253); font-size: 1.2rem; margin: 5px;"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
            />
          </svg>
        </a>
      </div>
      <div
        class="reservations-description"
      >
        <div
          class="reservations-header"
        >
          <h1>
            Book an Appointment
          </h1>
          <p>
            Select a doctor to make an appointment with.
          </p>
        </div>
        <select
          id="dropdown-menu-align-end"
          placeholder="Choose a Doctor you want to see."
          style="outline: none; width: 95%; border-radius: 2px;"
        >
          <option
            disabled=""
          >
            Choose a Doctor you want to see
          </option>
        </select>
        <div
          class="reservations-buttons"
        >
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container "
            >
              <span
                aria-live="polite"
                class="react-datepicker__aria-live"
                role="alert"
              />
              <input
                class=""
                type="text"
                value="04/26/2023"
              />
            </div>
          </div>
          <button
            class="book-btn"
            type="submit"
          >
            Book now
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
